Arelle/Arelle
# Merge ESEF and ESEF 2022 Plugins

#### Reason for change
Development on ESEF rule updates for 2023 is set to begin and we don't want to keep duplicating the ESEF plugin.

#### Description of change
* Merge `validate/ESEF_2022` plugin into `validate/ESEF` plugin using disclosure systems to select between the two (default is 2022).
* `validate/ESEF` plugin is updated to use PluginHooks base class.
* A bunch of code is deduplicated between the two plugins.
* Create replacement `ESEF_2022` plugin that informs users to migrate to `ESEF`.

PR is easier to review with whitespace changes ignored (?w=1).

#### Steps to Test
* ESEF CI
* Confirm the CLI with ESEF_2022 plugin directs you to use the ESEF plugin instead.
* Enable the `ESEF_2022` plugin (and disclosure system) with the master version of Arelle and confirm after upgrading to this version you're directed to use the ESEF plugin instead.

**review**:
@Arelle/arelle
@gmongelli
